web/client-api/src/main/java/io/deephaven/web/client/api/JsPartitionedTable.java

```diff
@@ -83,12 +88,18 @@ public Promise<JsPartitionedTable> refetch() {
             descriptor = PartitionedTableDescriptor.deserializeBinary(w.getDataAsU8());
 
             keyColumnTypes = new ArrayList<>();
+            InitialTableDefinition tableDefinition = WebBarrageUtils.readTableDefinition(
+                    WebBarrageUtils.readSchemaMessage(descriptor.getConstituentDefinitionSchema_asU8()));
             ColumnDefinition[] columnDefinitions = WebBarrageUtils.readColumnDefinitions(
                     WebBarrageUtils.readSchemaMessage(descriptor.getConstituentDefinitionSchema_asU8()));
-            for (int i = 0; i < columnDefinitions.length; i++) {
-                ColumnDefinition columnDefinition = columnDefinitions[i];
+            columns = new Column[0];
+            keyColumns = new Column[0];
+            for (ColumnDefinition columnDefinition : columnDefinitions) {
```

Comment:
java iterators don't translate as cleanly to patterns that make sense in JS, this is better left as a `for(int i =...` loop